### DISCUSSION TITLE
Do we stratify continuos target variables?

### DISCUSSION THREAD

**Discussion by kjappelbaum** [kjappelbaum/mofdscribe#242] [Ideas]
There is  *a lot* of debate around this, see, e.g. the [threads in sklearn](https://github.com/scikit-learn/scikit-learn/pull/14560).

Some thoughts: 
- this is already a weak source of data leakage as we need to consider the full dataset for binning. This is also mentioned in  [this stackexchange thread](https://stats.stackexchange.com/questions/117643/why-use-stratified-cross-validation-why-does-this-not-damage-variance-related-b):
> Stratified Cross-validation violates the principal that the test labels should never have been looked at before the statistics are calculated, but this is generally thought to be innocuous as the only effect is to balance the folds. Arguably the main reason stratification is important is to address defects in the model training process, as the model could be too easily biased by over- or under-representation of classes.
-  Some R-packages (including caret, or in [splittools](https://cran.r-project.org/web/packages/splitTools/vignettes/splitTools.html)) do it by default
- There is some debate (see sklearn discussions) about the best way of implementing: binning (e.g. `KBinsDiscretizer`) or simply [sorting and assigning every *n*th sample to a specific set](https://web.archive.org/web/20220128024145/https://scottclowe.com/2016-03-19-stratified-regression-partitions/). On top of that, there's also an interleaved or Venetian Blinds cross-validation approach that is implemented by the [pls](https://cran.r-project.org/web/packages/pls/vignettes/pls-manual.pdf) package. [Diago et al (2018)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5797612/)
 - A [comment from Max Kuhn](https://github.com/scikit-learn/scikit-learn/pull/14560#issuecomment-535204456) in the sklearn thread: 
>The text that @amueller was probably referring to was [here](https://bookdown.org/max/FES/data-splitting.html): 
	>  When the outcome is numeric, artificial strata can be constructed based on the quartiles of the data. For example, in the Ames housing price data, the quartiles of the outcome distribution would break the data into four artificial groups containing roughly 230 houses. The training/test split would then be conducted within these four groups and the four different training set portions are pooled together (and the same for the test set).
The idea is to make the resulting distributions of the target as close as possible (esp in the tails).
	> Indeed I am not sure about the value of our default choice either.
Unless your data set size is really small (say 2-10 samples per bin) I have never seen any harm in this (it is the default in caret). For larger data sets, the risk of getting discordant distributions is pretty low.
	> Also I think we could do a better on how to use the various options of our CV tools to detect breakage in the i.i.d. assumption.
I agree that with data sets with a strong time-effect, this could go badly (but any resampling method that doesn't account for seasonal/temporal patterns would be some level of problematic).
- There might be some more insight in [this paper](https://www.tandfonline.com/doi/abs/10.1080/09720510.2017.1411027?journalCode=tsms20). However, glancing through it, itself does not compare to unstratified splits.
